addons/gamification/models/gamification_goal_definition.py

```diff
@@ -62,6 +62,18 @@ class GamificationGoalDefinition(models.Model):
     action_id = fields.Many2one('ir.actions.act_window', string="Action", help="The action that will be called to update the goal value.")
     res_id_field = fields.Char("ID Field of user", help="The field name on the user profile (res.users) containing the value for res_id for action.")
 
+    @api.constrains("computation_mode", "model_id", "field_id")
+    def _check_computation_mode(self):
+        for definition in self:
+            if definition.computation_mode != "sum":
+                continue
+
+            if not definition.field_id:
```

Comment:
Would be great to include field name / definition name in error message (for batch update, knowing which one fails)